sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -0,0 +1,224 @@
+import {
+  ECERT_FULL_TIME_FILE_CODE,
+  ECERT_PART_TIME_FILE_CODE,
+  getDayOfTheYear,
+  getFieldOfStudyFromCIPCode,
+} from "../../utilities";
+import { EntityManager } from "typeorm";
+import { ESDCFileHandler } from "../esdc-file-handler";
+import { ECertUploadResult } from "./e-cert-full-time-integration/models/e-cert-full-time-integration.model";
+import { Injectable } from "@nestjs/common";
+import { Award, ECertRecord } from "./e-cert-integration-model";
+import { ECertIntegrationService } from "./e-cert-integration.service";
+import { ECertFullTimeIntegrationService } from "./e-cert-full-time-integration/e-cert-full-time-integration.service";
+import { ECertPartTimeIntegrationService } from "./e-cert-part-time-integration/e-cert-part-time-integration.service";
+
+const ECERT_FULL_TIME_SENT_FILE_SEQUENCE_GROUP = "ECERT_FT_SENT_FILE";
+const ECERT_PART_TIME_SENT_FILE_SEQUENCE_GROUP = "ECERT_PT_SENT_FILE";
+@Injectable()
+export class ECertFileHandler extends ESDCFileHandler {
+  constructor(
+    configService: ConfigService,
+    private readonly sequenceService: SequenceControlService,
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+    private readonly eCertFullTimeIntegrationService: ECertFullTimeIntegrationService,
+    private readonly eCertPartTimeIntegrationService: ECertPartTimeIntegrationService,
+  ) {
+    super(configService);
+  }
+
+  /**
+   * Method to call the Full-time disbursements available to be sent to ESDC.
+   * @returns result of the file upload with the file generated and the
+   * amount of records added to the file.
+   */
+  async generateFullTimeECert(): Promise<ECertUploadResult> {
+    return this.generateECert(
+      this.eCertFullTimeIntegrationService,
+      OfferingIntensity.fullTime,
+      ECERT_FULL_TIME_FILE_CODE,
+      ECERT_FULL_TIME_SENT_FILE_SEQUENCE_GROUP,
+    );
+  }
+
+  /**
+   * Method to call the Part-time disbursements available to be sent to ESDC.
+   * @returns result of the file upload with the file generated and the
+   * amount of records added to the file.
+   */
+  async generatePartTimeECert(): Promise<ECertUploadResult> {
+    return this.generateECert(
+      this.eCertPartTimeIntegrationService,
+      OfferingIntensity.partTime,
+      ECERT_PART_TIME_FILE_CODE,
+      ECERT_PART_TIME_SENT_FILE_SEQUENCE_GROUP,
+    );
+  }
+
+  /**
+   * Get all Full-Time/ Part-Time disbursements available to be sent to ESDC.
+   * Consider any record that is scheduled in upcoming days or in the past.
+   * @param eCertIntegrationService
+   * @param offeringIntensity disbursement offering intensity.
+   * @param fileCode File code applicable for Part-Time or Full-Time.
+   * @param sequenceGroup Sequence group application for Part-Time or Full-Time.
+   * @returns result of the file upload with the file generated and the
+   * amount of records added to the file.
+   */
+  async generateECert(
+    eCertIntegrationService: ECertIntegrationService<any>,
```

Comment:
I believe that any can be typped as below for now while we do not have the response for part-time.
```typescript
eCertIntegrationService: ECertIntegrationService<ECertResponseRecord[] | void>
```